tests/test_auth.py

```diff
@@ -0,0 +1,191 @@
+import os
+import tempfile
+from unittest.mock import patch
+
+import pytest
+
+from huggingface_hub import constants
+from huggingface_hub._login import _login, _set_active_token, auth_switch, logout
+from huggingface_hub.utils._auth import (
+    _get_token_by_name,
+    _get_token_from_file,
+    _save_token,
+    get_stored_tokens,
+)
+
+from .testing_constants import ENDPOINT_STAGING, OTHER_TOKEN, TOKEN
+
+
+@pytest.fixture(autouse=True)
+def use_tmp_file_paths():
+    """
+    Fixture to temporarily override HF_TOKEN_PATH, HF_STORED_TOKENS_PATH, and ENDPOINT.
+
+    This fixture patches the constants in the huggingface_hub module to use the
+    specified paths and the staging endpoint. It also ensures that the files are
+    deleted after all tests in the module are completed.
+    """
+    with tempfile.TemporaryDirectory() as tmp_hf_home:
+        hf_token_path = os.path.join(tmp_hf_home, "token")
+        hf_stored_tokens_path = os.path.join(tmp_hf_home, "stored_tokens")
+        with patch.multiple(
+            constants,
+            HF_TOKEN_PATH=hf_token_path,
+            HF_STORED_TOKENS_PATH=hf_stored_tokens_path,
+            ENDPOINT=ENDPOINT_STAGING,
+        ):
+            yield
+
+
+class TestGetTokenByName:
+    def test_get_existing_token(self):
+        _save_token(TOKEN, "test_token")
+        token = _get_token_by_name("test_token")
+        assert token == TOKEN
+
+    def test_get_non_existent_token(self):
+        assert _get_token_by_name("non_existent") is None
+
+
+class TestSaveToken:
+    def test_save_new_token(self):
+        _save_token(TOKEN, "new_token")
+
+        stored_tokens = get_stored_tokens()
+        assert "new_token" in stored_tokens
+        assert stored_tokens["new_token"] == TOKEN
+
+    def test_overwrite_existing_token(self):
+        _save_token(TOKEN, "test_token")
+        _save_token("new_token", "test_token")
+
+        assert _get_token_by_name("test_token") == "new_token"
+
+
+class TestSetActiveToken:
+    def test_set_active_token_success(self):
+        _save_token(TOKEN, "test_token")
+        _set_active_token("test_token", add_to_git_credential=False)
+        assert _get_token_from_file() == TOKEN
+
+    def test_set_active_token_non_existent(self):
+        non_existent_token = "non_existent"
+        with pytest.raises(
+            ValueError, match=rf"Token {non_existent_token} not found in {constants.HF_STORED_TOKENS_PATH}"
```

Comment:
```suggestion
            ValueError, match="Token non_existent not found in .*"
```

this should fix the CI failure on Windows and is more than enough to test the ValueError in case of non existent